src/services/charmcraft/charmcraft.ts

```diff
@@ -164,16 +164,29 @@ class Charmcraft {
     };
   }
 
-  _read() {
+  _readMetadata(): Metadata {
```

Comment:
this is typescript, we should use `private readMetadata` rather than `_readMetadata`

Reply:
Gotcha, updated, thanks!